pkg/docker/config/config.go

```diff
@@ -538,17 +560,32 @@ func getPathToAuthWithOS(sys *types.SystemContext, goOS string) (string, bool, e
 // readJSONFile unmarshals the authentications stored in the auth.json file and returns it
 // or returns an empty dockerConfigFile data structure if auth.json does not exist
 // if the file exists and is empty, readJSONFile returns an error
-func readJSONFile(path string, legacyFormat bool) (dockerConfigFile, error) {
+func readJSONFile(path string, legacyFormat, requireUserOnly bool) (dockerConfigFile, error) {
 	var auths dockerConfigFile
 
-	raw, err := os.ReadFile(path)
+	f, err := os.Open(path)
 	if err != nil {
 		if os.IsNotExist(err) {
 			auths.AuthConfigs = map[string]dockerAuthConfig{}
 			return auths, nil
 		}
 		return dockerConfigFile{}, err
 	}
+	defer f.Close()
+	if requireUserOnly {
+		st, err := f.Stat()
+		if err != nil {
+			return dockerConfigFile{}, fmt.Errorf("stat %s: %w", path, err)
+		}
+		perms := st.Mode().Perm()
+		if (perms & 04) > 0 {
+			return dockerConfigFile{}, fmt.Errorf("refusing to process %s with world read permissions", path)
```

Comment:
For now I chose to error on group/world read permissions.  Forcing 0600 also forces write permissions, but I can imagine cases where one might not want that.  (Not that it really matters because root usually has `CAP_DAC_OVERRIDE`)